simple_salesforce/api.py

```diff
@@ -207,6 +207,12 @@ def __init__(
 
         self.api_usage = {}
 
+        try:
+            self._sobject_names = [x['name'] for x
+                                   in self.describe()['sobjects']]
+        except: # pylint: disable=bare-except
+            self._sobject_names = []
+
```

Comment:
Maybe this code could be moved inside the `__dir__` function, so it does not run in production code. Also not sure the [bare except](https://stackoverflow.com/a/4992124/2654518) is a good idea -- could be at least `except Exception:`.